locale/translators/index.php

```diff
@@ -561,3 +564,41 @@ function validate_locale($locale, $check_dir_exists = true)
     }
     return $locale;
 }
+
+function build_translation_js()
+{
+    global $dyn_locales_dir;
+
+    $locale_json = [];
+    foreach (get_installed_locale_translations() as $locale) {
+        $po_filename = "$dyn_locales_dir/$locale/LC_MESSAGES/messages.po";
+        if (!is_locale_translation_enabled($locale)) {
+            continue;
+        }
+
+        if (!$tempfile = tempnam(sys_get_temp_dir(), "translation_js")) {
+            throw new RuntimeException("Unable to create temporary file");
+        }
+        // TODO: Instead of including the entire messages file, just pull out
+        // the strings used in *.js files (unless the files doesn't have any
+        // JS files in which case fall back to loading them all).
+        $pofile = new POFile($po_filename);
+        $pofile->convert_to_json($tempfile);
+        $locale_json[str_replace("_", "-", $locale)] = json_decode(file_get_contents($tempfile));
+        unlink($tempfile);
+    }
+
+    // use "var" not "const" because Safari <12 can't see those within modules
+    // due to some scoping weirdness
```

Comment:
can we just bump safari version? Also curious about the bug, but understand if you don't have a webkit bug number or anything.

Reply:
The comment is wrong and I need to update it, `const` didn't work until Safari 14, not Safari 12, which is a pretty big jump for this change. I don't have a bug for it -- just tracked it to a Safari behavior change via BrowserStack.